lib/puppet/provider/oneview_network_set/c7000.rb

```diff
@@ -14,30 +14,21 @@
 # limitations under the License.
 ################################################################################
 
-require_relative '../login'
-require_relative '../common'
-require 'oneview-sdk'
+require_relative '../oneview_resource'
 
-Puppet::Type::Oneview_network_set.provide :c7000 do
+Puppet::Type::Oneview_network_set.provide :c7000, parent: Puppet::OneviewResource do
   desc 'Provider for OneView Network Sets using the C7000 variant of the OneView API'
 
   mk_resource_methods
 
   def initialize(*args)
     super(*args)
-    @client = OneviewSDK::Client.new(login)
-    api_version = login[:api_version] || 200
-    @resourcetype ||= if api_version == 200
-                        OneviewSDK::API200::NetworkSet
-                      else
-                        Object.const_get("OneviewSDK::API#{api_version}::C7000::NetworkSet")
-                      end
+    api_version ||= login[:api_version] || 200
     @ethernet ||= if api_version == 200
```

Comment:
Maybe "ethernet" could be changed by "ethernet_client" or something more specific.